app/upgrades/v17/concentrated_pool.go

```diff
@@ -0,0 +1,55 @@
+package v17
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+
+	v16 "github.com/osmosis-labs/osmosis/v17/app/upgrades/v16"
+	clmodel "github.com/osmosis-labs/osmosis/v17/x/concentrated-liquidity/model"
+	gammkeeper "github.com/osmosis-labs/osmosis/v17/x/gamm/keeper"
+	"github.com/osmosis-labs/osmosis/v17/x/poolmanager"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v17/x/poolmanager/types"
+)
+
+// createConcentratedPoolFromCFMM creates a new concentrated liquidity pool with the desiredDenom0 token as the
+// token 0, links it with an existing CFMM pool, and returns the created pool.
+// It uses pool manager module account as the creator of the pool.
+// Returns error if desired denom 0 is not in associated with the CFMM pool.
+// Returns error if CFMM pool does not have exactly 2 denoms.
+// Returns error if pool creation fails.
+func createConcentratedPoolFromCFMM(ctx sdk.Context, cfmmPoolIdToLinkWith uint64, desiredDenom0 string, spreadFactor sdk.Dec, accountKeeper authkeeper.AccountKeeper, gammKeeper gammkeeper.Keeper, poolmanagerKeeper poolmanager.Keeper) (poolmanagertypes.PoolI, error) {
```

Comment:
note: this was repetetive across different part of the codebase so i made it a part of gamm keeper here (https://github.com/osmosis-labs/osmosis/pull/5890/files#diff-5bab9ed33d94273b77a7c0f5913ee0f8dd8350ee1842025c64fab3a7d8b98dabR367) maybe we should do that here

Reply:
This one does not auto pull the spread factor since we can't be sure that the gamm's spread factor is authorized. I will add this version to migrate.go though in place of the auto pull version

Reply:
Followed similar to what you did in your PR but changed it to take the the spreadFactor as an input rather than pulling it from the previous pool.